go/geth_adapter/adapter_test.go

```diff
@@ -75,3 +86,311 @@ func TestRunContextAdapter_SetBalanceHasCorrectEffect(t *testing.T) {
+	stateDb := NewMockStateDb(ctrl)
+	adapter := &runContextAdapter{evm: &geth.EVM{StateDB: stateDb}}
+
+	snapshot := tosca.Snapshot(1)
+
+	stateDb.EXPECT().Snapshot().Return(int(snapshot))
+	adapter.CreateSnapshot()
+
+	stateDb.EXPECT().RevertToSnapshot(int(snapshot))
+	adapter.RestoreSnapshot(snapshot)
+}
+
+func TestRunContextAdapter_AccountOperations(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateDb := NewMockStateDb(ctrl)
+	adapter := &runContextAdapter{evm: &geth.EVM{StateDB: stateDb}}
+
+	address := tosca.Address{0x42}
+	code := tosca.Code{1, 2, 3}
+
+	stateDb.EXPECT().AddressInAccessList(gc.Address(address))
+	stateDb.EXPECT().AddAddressToAccessList(gc.Address(address))
+	adapter.AccessAccount(address)
+
+	stateDb.EXPECT().Exist(gc.Address(address)).Return(true)
+	adapter.AccountExists(address)
+
+	stateDb.EXPECT().Exist(gc.Address(address)).Return(false)
+	stateDb.EXPECT().CreateAccount(gc.Address(address))
+	stateDb.EXPECT().SetCode(gc.Address(address), code)
+	adapter.CreateAccount(address, code)
+
+	stateDb.EXPECT().AddressInAccessList(gc.Address(address))
+	adapter.IsAddressInAccessList(address)
+}
+
+type testContractRef struct {
+	address gc.Address
+}
+
+func (c testContractRef) Address() gc.Address {
+	return c.address
+}
+
+func TestRunContextAdapter_Call(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	stateDb := NewMockStateDb(ctrl)
+
+	address := gc.Address{0x42}
+
+	stateDb.EXPECT().Snapshot().Return(1)
+	stateDb.EXPECT().Exist(address).Return(true)
+	stateDb.EXPECT().GetCode(address).Return([]byte{})
+	stateDb.EXPECT().Witness()
+
+	contractRef := testContractRef{address: address}
+
+	canTransfer := func(geth.StateDB, gc.Address, *uint256.Int) bool { return true }
+	transfer := func(geth.StateDB, gc.Address, gc.Address, *uint256.Int) {}
+
+	runContextAdapter := &runContextAdapter{
+		evm: &geth.EVM{
+			StateDB: stateDb,
+			Context: geth.BlockContext{
+				CanTransfer: canTransfer,
+				Transfer:    transfer,
+			},
+		},
+		contract: geth.NewContract(contractRef, contractRef, nil, 0),
+	}
+
+	gas := tosca.Gas(42)
+
+	parameters := tosca.CallParameters{
+		Recipient: tosca.Address(address),
+		Gas:       gas,
+	}
+
+	result, err := runContextAdapter.Call(tosca.Call, parameters)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	if !result.Success {
+		t.Errorf("Call was not successful")
+	}
+	if result.GasLeft != gas {
+		t.Errorf("Call has the wrong amount of gas left: %v, expected: %v", result.GasLeft, gas)
+	}
+}
+
+func TestRunContextAdapter_Run(t *testing.T) {
+	tests := map[string]bool{
+		"success": true,
+		"failure": false,
+	}
+
+	for name, success := range tests {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			stateDb := NewMockStateDb(ctrl)
+			interpreter := tosca.NewMockInterpreter(ctrl)
+
+			refundShift := uint64(1 << 60)
+			stateDb.EXPECT().AddRefund(refundShift)
+
+			interpreter.EXPECT().Run(gomock.Any()).Return(tosca.Result{Success: success}, nil)
```

Comment:
This expectation should cover some of the values of the expected parameters. In particular, it should check that the `static` flag is properly set, as well as things like gas, code, value, codehash, ...